tests/groupRole.test.js

```diff
@@ -0,0 +1,1349 @@
+const ERR = require('async-stacktrace');
+
+const config = require('../lib/config');
+const path = require('path');
+var assert = require('chai').assert;
+var request = require('request');
+var cheerio = require('cheerio');
+
+var sqldb = require('@prairielearn/postgres');
+var sql = sqldb.loadSqlEquiv(__filename);
+
+var helperServer = require('./helperServer');
+
+let page, elemList;
+const locals = {};
+locals.helperClient = require('./helperClient');
+locals.siteUrl = 'http://localhost:' + config.serverPort;
+locals.baseUrl = locals.siteUrl + '/pl';
+locals.courseInstanceUrl = locals.baseUrl + '/course_instance/1';
+locals.assessmentsUrl = locals.courseInstanceUrl + '/assessments';
+locals.courseDir = path.join(__dirname, '..', 'testCourse');
+
+const storedConfig = {};
+
+describe('Test group based assessments with custom group roles from student side', function () {
+  this.timeout(20000);
+  before('set authenticated user', function (callback) {
+    storedConfig.authUid = config.authUid;
+    storedConfig.authName = config.authName;
+    storedConfig.authUin = config.authUin;
+    callback(null);
+  });
+  before('set up testing server', helperServer.before(locals.courseDir));
+  after('shut down testing server', helperServer.after);
+  after('unset authenticated user', function (callback) {
+    Object.assign(config, storedConfig);
+    callback(null);
+  });
+
+  describe('1. the database', function () {
+    it('should contain a group-based homework assessment with roles', function (callback) {
```

Comment:
The function passed here should be `async` instead of using callbacks; ditto everywhere else in this file.